build.gradle

```diff
@@ -166,6 +166,86 @@ subprojects.stream().filter { it.parent != rootProject }.each { p ->
 
         sourceSets {
             main.kotlin.srcDirs += "$projectDir/src"
+            test.kotlin.srcDirs += "$projectDir/test"
+
+            testPlain {
+                kotlin.srcDirs += "$projectDir/test"
+                compileClasspath = test.compileClasspath
+            }
+
+            testCommonJS {
+                kotlin.srcDirs += "$projectDir/test"
+                compileClasspath = test.compileClasspath
+            }
+
+            testUmdOnly {
+                kotlin.srcDirs += "$projectDir/testUMD"
+                compileClasspath = test.compileClasspath
+            }
+
+            testPlainOnly {
+                kotlin.srcDirs += "$projectDir/testPlain"
+                compileClasspath = test.compileClasspath
+            }
+
+            testCommonJsOnly {
+                kotlin.srcDirs += "$projectDir/testCommonJS"
+                compileClasspath = test.compileClasspath
+            }
+
+            // Shared
+
+            testSharedUMD {
+                kotlin.srcDirs += "$projectDir/../_testShared"
+                compileClasspath = test.compileClasspath
+            }
+
+            testSharedPlain {
+                kotlin.srcDirs += "$projectDir/../_testShared"
+                compileClasspath = test.compileClasspath
+            }
+
+            testSharedCommonJS {
+                kotlin.srcDirs += "$projectDir/../_testShared"
+                compileClasspath = test.compileClasspath
+            }
+
+            testSharedUmdOnly {
+                kotlin.srcDirs += "$projectDir/../_testSharedUMD"
+                compileClasspath = test.compileClasspath
+            }
+
+            testSharedPlainOnly {
+                kotlin.srcDirs += "$projectDir/../_testSharedPlain"
+                compileClasspath = test.compileClasspath
+            }
+
+            testSharedCommonJsOnly {
+                kotlin.srcDirs += "$projectDir/../_testSharedCommonJS"
+                compileClasspath = test.compileClasspath
+            }
```

Comment:
Why not just add "$projectDir/../_testSharedCommonJS" to testSharedCommonJS?

Reply:
Done